component/src/test/java/org/wso2/carbon/identity/authenticator/pinterest/PinterestTest.java

```diff
@@ -0,0 +1,199 @@
+import org.testng.annotations.ObjectFactory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.exception.ApplicationAuthenticatorException;
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.exception.InvalidCredentialsException;
+import org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import org.wso2.carbon.identity.application.common.model.ClaimMapping;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({OAuthAuthzResponse.class, OAuthClientRequest.class})
+public class PinterestTest extends PowerMockTestCase {
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    private PinterestAuthenticator pinterestAuthenticator;
+    @Mock
+    private HttpServletRequest httpServletRequest;
+    @Mock
+    private HttpServletResponse httpServletResponse;
+    @Mock
+    private Map<String, String> authenticatorProperties;
+    @Mock
+    private OAuthClientResponse oAuthClientResponse;
+    @Mock
+    private AuthenticationContext authenticationContext;
+    @Mock
+    private OAuthAuthzResponse authAuthzResponse;
+    @Mock
+    private OAuthClient oAuthClient;
+    @Mock
+    private BufferedReader bufferedReader;
+
+    @Mock
+    private OAuthClient mockOAuthClient;
+
+    @Mock
+    private OAuthClientRequest mockOAuthClientRequest;
+    private OAuthJSONAccessTokenResponse oAuthJSONAccessTokenResponse = new OAuthJSONAccessTokenResponse();
+    private Map<ClaimMapping, String> map = new HashMap<>();
+
+    @Spy
+    private AuthenticationContext context = new AuthenticationContext();
+
+    @DataProvider
+    public Object[][] getAuthenticatorPropertiesData() {
+
+        Map<String, String> authenticatorProperties = new HashMap<>();
+        authenticatorProperties.put(OIDCAuthenticatorConstants.CLIENT_ID, "test-client-id");
+        authenticatorProperties.put(OIDCAuthenticatorConstants.CLIENT_SECRET, "test-client-secret");
+        authenticatorProperties.put("callbackUrl", "http://localhost:9443/commonauth");
+        return new Object[][]{{authenticatorProperties}};
+    }
+
+    @BeforeMethod
+    public void setUp() {
+        pinterestAuthenticator = new PinterestAuthenticator();
+        initMocks(this);
+    }
+
+    @Test
```

Comment:
Add test descriptions to the missing test cases.

Reply:
fixed